snet_cli/identity.py

```diff
@@ -47,8 +47,8 @@ def __init__(self, w3, private_key):
                                                   curve=ecdsa.SECP256k1,
                                                   hashfunc=hashlib.sha256).get_verifying_key()
 
-        self.address = self.w3.toChecksumAddress(
-            "0x" + self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex())
+        self.address = self.w3.sha3(hexstr=public_key.to_string().hex())[12:].hex()
+        self.address = self.w3.toChecksumAddress(self.address) if self.address.startswith("0x") else self.w3.toChecksumAddress("0x" + self.address)
```

Comment:
You probably should directly use web3.eth.accounts.privateKeyToAccount.

Reply:
That's JavaScript: https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#privatekeytoaccount

I couldn't find the same in Python.

Reply:
@vforvalerio87 it took 30sec to find web3py alternative!
```python
import web3
print(web3.eth.Account.privateKeyToAccount(self.private_key).address)
```
please use it

Reply:
(it is actually not so easy to find this function in web3py. I found it in the source code of web3py. But I think we should use it... )